src/platform_impl/linux/x11/window.rs

```diff
@@ -1702,7 +1709,28 @@ impl UnownedWindow {
 
     #[inline]
     pub fn request_activation_token(&self) -> Result<AsyncRequestSerial, NotSupportedError> {
-        Err(NotSupportedError::new())
+        // Get the title from the WM_NAME property.
+        let atoms = self.xconn.atoms();
+        let title = {
+            let title_bytes = self
+                .xconn
+                .get_property(self.xwindow, atoms[_NET_WM_NAME], atoms[UTF8_STRING])
+                .expect("Failed to get WM_NAME property");
+
+            String::from_utf8(title_bytes).map_err(|_| NotSupportedError::new())?
+        };
+
+        // Get the activation token and then put it in the event queue.
+        let token = self
+            .xconn
+            .request_activation_token(&title)
```

Comment:
Maybe we should request the token right before sending it back to the user?

Reply:
Good idea, I've implemented this.